From 7971ec157a8451c684845e33a5a7b2502b5c1a27 Mon Sep 17 00:00:00 2001
From: Madalin Bucur <madalin.bucur@freescale.com>
Date: Fri, 3 Jan 2014 13:28:29 +0200
Subject: [PATCH 014/466] dpaa_eth: fix dependency on
 fm_port_suspend()/fm_port_resume()

Fix compiler error caused by missing autoresponse code from
FMD. The AR code is not compatible with newer kernel API and
an updated patch is not yet available. Bypassing
fm_port_suspend()/fm_port_resume() calls when FMD support
for AR is not present.

Signed-off-by: Madalin Bucur <madalin.bucur@freescale.com>
Change-Id: I21b1e6014dabed1b3fd0d83b098debcd7bf9c057
Reviewed-on: http://git.am.freescale.net:8181/7664
[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h
index 9e82fe7..4746a34 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_common.h
@@ -37,6 +37,17 @@
 #include <linux/of_platform.h> /* struct platform_device */
 #include <linux/net_tstamp.h>	/* struct hwtstamp_config */
 
+/* make dpaa_eth code compile without the FMD AR code */
+#ifdef CONFIG_PM
+#ifndef fm_port_suspend
+#define fm_port_suspend(x) (0)
+#warning fm_port_suspend missing
+#endif
+#ifndef fm_port_resume
+#define fm_port_resume(x) (0)
+#endif
+#endif /* CONFIG_PM */
+
 #define dpaa_eth_init_port(type, port, param, errq_id, defq_id, buf_layout,\
 			   frag_enabled) \
 { \
-- 
1.7.10.4

